packages/container/libraries/strongly-typed/src/container.ts

```diff
@@ -0,0 +1,210 @@
+  >(
+    serviceIdentifier: TKey,
+  ) => Promise<TBound[]>;
+  getAllTaggedAsync: <
+    TBound extends ContainerBinding<T, TKey>,
+    TKey extends MappedServiceIdentifier<T> = any,
+  >(
+    serviceIdentifier: TKey,
+    key: PropertyKey,
+    value: unknown,
+  ) => Promise<TBound[]>;
+  getAllNamedAsync: <
+    TBound extends ContainerBinding<T, TKey>,
+    TKey extends MappedServiceIdentifier<T> = any,
+  >(
+    serviceIdentifier: TKey,
+    named: PropertyKey,
+  ) => Promise<TBound[]>;
+  isBound: IsBound<T>;
+  isBoundNamed: (
+    serviceIdentifier: MappedServiceIdentifier<T>,
+    named: PropertyKey,
+  ) => boolean;
+  isBoundTagged: (
+    serviceIdentifier: MappedServiceIdentifier<T>,
+    key: PropertyKey,
+    value: unknown,
+  ) => boolean;
+  isCurrentBound: IsBound<T>;
+  rebind: Rebind<T>;
+  rebindAsync: RebindAsync<T>;
+  unbind: Unbind<T>;
+  unbindAsync: UnbindAsync<T>;
+  onActivation<
+    TBound extends ContainerBinding<T, TKey>,
+    TKey extends MappedServiceIdentifier<T> = any,
+  >(
+    serviceIdentifier: TKey,
+    onActivation: interfaces.BindingActivation<TBound>,
+  ): void;
+  onDeactivation<
+    TBound extends ContainerBinding<T, TKey>,
+    TKey extends MappedServiceIdentifier<T> = any,
+  >(
+    serviceIdentifier: TKey,
+    onDeactivation: interfaces.BindingDeactivation<TBound>,
+  ): void;
+  resolve<TBound>(constructorFunction: interfaces.Newable<TBound>): TBound;
+  createChild<TChild extends BindingMap = any>(
+    containerOptions?: interfaces.ContainerOptions,
+  ): TypedContainer<TChild, [T, ...TAncestors]>;
+}
+
+type Bind<T extends BindingMap = any> = <
+  TBound extends ContainerBinding<T, TKey>,
+  TKey extends MappedServiceIdentifier<T> = any,
+>(
+  serviceIdentifier: TKey,
+) => interfaces.BindingToSyntax<TBound>;
+
+type Rebind<T extends BindingMap = any> = Bind<T>;
+
+type RebindAsync<T extends BindingMap = any> = <
+  TBound extends ContainerBinding<T, TKey>,
+  TKey extends MappedServiceIdentifier<T> = any,
+>(
+  serviceIdentifier: TKey,
+) => Promise<interfaces.BindingToSyntax<TBound>>;
+
+type Unbind<T extends BindingMap = any> = <
+  TKey extends MappedServiceIdentifier<T>,
+>(
+  serviceIdentifier: TKey,
+) => void;
+
+type UnbindAsync<T extends BindingMap = any> = <
+  TKey extends MappedServiceIdentifier<T>,
+>(
+  serviceIdentifier: TKey,
+) => Promise<void>;
+
+type IsBound<T extends BindingMap = any> = <
+  TKey extends MappedServiceIdentifier<T>,
+>(
+  serviceIdentifier: TKey,
+) => boolean;
+
+export type TypedContainer<
+  T extends BindingMap = any,
+  TAncestors extends BindingMap[] = any[],
+> = ContainerOverrides<T, TAncestors> &
+  Omit<interfaces.Container, keyof ContainerOverrides>;
+
+// eslint-disable-next-line @typescript-eslint/naming-convention, @typescript-eslint/no-unsafe-assignment
+export const TypedContainer: {
+  new <T extends BindingMap = any, TAncestors extends BindingMap[] = any[]>(
+    ...args: any[]
```

Comment:
```suggestion
    containerOptions?: interfaces.ContainerOptions,
```
Let's keep the original constructor params